tests/test_selective_validation.py

```diff
@@ -41,7 +41,10 @@ def test_wrapped_class(all_arg_names, selective_arg_names, shape, origin, domain
             with no nans
     """
     origin_domain_func = MagicMock(return_value=(origin, domain))
-    Wrapped = get_selective_class(DummyClass, selective_arg_names, origin_domain_func)
+    name_to_function = {}
```

Comment:
Can you add tests of the new feature? This may be easiest as a new test, because you need two different origin_domain_func's and you need to look at `origin_domain_func.call_count` to make sure that each one is called once. You don't need to cover al the origin/domain/argument combinations covered by this existing test, one case should be enough.